### PR TITLE
[WIP] prototype use of tabular json response

### PR DESCRIPTION
**Pull request by juanvallejo**
This patch prototypes the use of the "server-side GET operation"
introduced by pull/40848 and proposed by kubernetes/community/pull/363.

This is simply a demonstration of how it could eventually be implemented
in "kubectl get".

As Jordan has pointed out, there are a few pieces missing on the
server-side before this feature can be fully used, such as fallback to
a standard JSON response in the event that a tabular response for a
specific resource is not supported. Additionally, I am not yet able to 
request `TableLists` as part of the "Accept" header.

Feedback is welcome, as well as further discussion on the future
implementation of this feature.

**Release note**:
```release-note
NONE
```

cc @fabianofranz @liggitt @deads2k @smarterclayton

### CODE REVIEW

**Comment by juanvallejo** [comment]
Sample use:

```
$ kubectl get pod --experimental-use-server-print-columns
Flag --experimental-use-server-print-columns has been deprecated, This flag is experimental and may be removed in the future.
NAME          READY     STATUS    RESTARTS   AGE
simple-pod2   0/1       Pending   0          2h

```

**Comment by deads2k** [comment]
@smarterclayton Looks like it kind of works.  I might wire it slightly differently, but this is the first proof of working that I've seen.

File: pkg/kubectl/resource/builder.go [hunk 1]
```diff
@@ -169,6 +170,13 @@ func (b *Builder) Local(mapperFunc ClientMapperFunc) *Builder {
 	return b
 }
 
+// TabularResponse sets the header "Accept: application/json;as=Table;v=v1alpha1;g=meta.k8s.io"
+// as part of a request to a resource endpoint
+func (b *Builder) TabularResponse(tabular bool) *Builder {
```
Comment (smarterclayton) [inline pkg/kubectl/resource/builder.go]: This isn't going to work well with unstructured builders.  I don't know that the builder is responsible for knowing about tables.  At some point we're going to have commands that deal with generic metadata, and those shouldn't get their own methods.  I think this is a lower level detail that needs to be handled more cleanly.

**Pull request closed without merging**